examples/calibrate_tsqrtP.py

```diff
@@ -0,0 +1,212 @@
+r"""
+Calculate actual beta as a function of pulse length
+===================================================
+Assuming the data is the capture of the pulse sequence as seen on the GDS
+oscilloscope (acquired using FLInst/examples/calib_pulses.py), 
+here the data is converted to analytic power, frequency filtered
+and the absolute is taken prior to integrating to return the beta where
+:math:`\beta = \sqrt{2} \int \sqrt{P(t)} dt` 
+"""
+import pyspecdata as psd
+import matplotlib.pyplot as plt
+import numpy as np
+from itertools import cycle
+
+colorcyc_list = plt.rcParams["axes.prop_cycle"].by_key()["color"]
+color_cycle = cycle(
+    colorcyc_list
+)  # this can be done more than once to spin up multiple lists
+
+V_atten_ratio = 102.35  # attenutation ratio
+skip_plots = 33  # diagnostic -- set this to None, and there will be no plots
+linear_threshold = 100e-6
+
+
+def prog_plen(desired):
+    """function that takes the coefficients of the linear and nonlinear
+    regions and applies the fit respectively to calculate the pulse time that
+    will return the desired beta value
+    """
+
+    def zonefit(desired):
+        if desired > linear_threshold:
+            return np.polyval(c_linear[::-1], desired)
+        else:
+            return np.polyval(c_nonlinear[::-1], desired)
+
+    ret_val = np.vectorize(zonefit)(desired)
+    if ret_val.size > 1:
+        return ret_val
+    else:
+        return ret_val.item()
+
+
+with psd.figlist_var() as fl:
+    for filename, nodename in [
+        ("240805_calib_amp1_pulse_calib.h5", "pulse_calib_1"),  # high power
+        ("240805_calib_amp0p1_a_pulse_calib.h5", "pulse_calib_3"),  # low power
+        ("240805_calib_amp0p2_a_pulse_calib.h5", "pulse_calib_1"),  # low power
+        ("240805_calib_amp0p05_pulse_calib.h5", "pulse_calib_5"),  # low power
+    ]:
+        d = psd.find_file(
+            filename, expno=nodename, exp_type="ODNP_NMR_comp/test_equipment"
+        )
```

Comment:
the ppg doesn't have a postproc type.  You should set a postproc_type and add an assert statement here

you don't have to do anything, but want to make sure you understand the data that you're reading

Reply:
for existing dataset, you can hack in the postproc type (there is an example of how to do this or something similar for the coherence pathway on slack)